### PR TITLE
Review Criteo Examples and Added Unittest

### PR DESCRIPTION
**Pull request by bschifferer**
I reviewed the criteo examples and made sure, they run successfully.
I updated descriptions
I added unittest for hugectr notebooks

### CODE REVIEW

File: examples/scaling-criteo/README.md [hunk 1]
```diff
@@ -13,9 +13,14 @@ Our recommendation is to use our latest stable [Merlin containers](https://catal
 
 Explore the following notebooks:
 
+Training and Deployment with **TensorFlow**:
+- [Download and Convert](01-Download-Convert.ipynb)
+- [Feature Engineering with NVTabular](02-ETL-with-NVTabular.ipynb)
+- [Training with HugeCTR](03-Training-with-HugeCTR.ipynb)
+- [Deploy the HugeCTR Model with Triton Inference Server](04-Triton-Inference-with-HugeCTR.ipynb)
```
Comment (mikemckiernan) [inline examples/scaling-criteo/README.md:20]: Should these be the TF notebooks rather than the HugeCTR ones?

**Review by mikemckiernan — APPROVED** [review]
A handful of nits that are up to you.  Thanks very much and let me know if I can clarify anything.

File: examples/scaling-criteo/03-Training-with-HugeCTR.ipynb [hunk 2]
```diff
@@ -78,27 +78,22 @@
    ]
```
Comment (radekosmulski) [inline examples/scaling-criteo/03-Training-with-HugeCTR.ipynb:78]: <p>written in C++ ('in' missing)</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/NVIDIA-Merlin/Merlin/pull/756/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='1'/>

**Review by radekosmulski — APPROVED** [review]

File: examples/scaling-criteo/01-Download-Convert.ipynb [hunk 3]
```diff
@@ -43,7 +43,7 @@
    "source": [
```
[the remark this answers is not in the document]
Comment (bschifferer) [reply examples/scaling-criteo/01-Download-Convert.ipynb:43]: I removed the step 1 to step n

File: examples/scaling-criteo/02-ETL-with-NVTabular.ipynb [hunk 4]
```diff
@@ -286,15 +526,15 @@
     "### Defining our Preprocessing Pipeline\n",
     "At this point, our data still isn't in a form that's ideal for consumption by neural networks. The most pressing issues are missing values and the fact that our categorical variables are still represented by random, discrete identifiers, and need to be transformed into contiguous indices that can be leveraged by a learned embedding. Less pressing, but still important for learning dynamics, are the distributions of our continuous variables, which are distributed across multiple orders of magnitude and are uncentered (i.e. E[x] != 0).\n",
     "\n",
-    "We can fix these issues in a concise and GPU-accelerated manner with an NVTabular `Workflow`. We explained the NVTabular API in [Getting Started with Movielens notebooks](https://github.com/NVIDIA/NVTabular/tree/main/examples/getting-started-movielens) and hope you are familiar with the syntax.\n",
+    "We can fix these issues in a concise and GPU-accelerated manner with an NVTabular `Workflow`. More information about NVTabular's API can be found in its [repository](https://nvidia-merlin.github.io/NVTabular/main/Introduction.html)\n",
```
[the remark this answers is not in the document]
Comment (bschifferer) [reply examples/scaling-criteo/02-ETL-with-NVTabular.ipynb:529]: changed to documentation

File: examples/scaling-criteo/03-Training-with-HugeCTR.ipynb [hunk 5]
```diff
@@ -118,35 +113,24 @@
...
+    "os.system(\"rm -rf \" + os.path.join(OUTPUT_DATA_DIR, \"criteo_hugectr/\"))\n",
+    "os.system(\"mkdir -p \" + os.path.join(OUTPUT_DATA_DIR, \"criteo_hugectr/1/\"))"
    ]
   },
   {
    "cell_type": "markdown",
    "metadata": {},
    "source": [
-    "We use `graph_to_json` to convert the model to a JSON configuration, required for the inference."
+    "We will write the code to a `./model.py` file and execute it. It will create snapshot, which we will use for inference in the next notebook. We use `graph_to_json` to convert the model to a JSON configuration, required for the inference."
```
[the remark this answers is not in the document]
Comment (bschifferer) [reply examples/scaling-criteo/03-Training-with-HugeCTR.ipynb:124]: updated thanks

File: examples/scaling-criteo/04-Triton-Inference-with-HugeCTR.ipynb [hunk 6]
```diff
@@ -140,44 +136,50 @@
...
    "source": [
-    "os.system(\"rm -rf /model/*\")"
+    "os.system(\"rm -rf \" + os.path.join(OUTPUT_DATA_DIR, \"model_inference\"))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "#### Export artifcats\n",
```
[the remark this answers is not in the document]
Comment (bschifferer) [reply examples/scaling-criteo/04-Triton-Inference-with-HugeCTR.ipynb:150]: updated thanks

File: examples/scaling-criteo/04-Triton-Inference-with-HugeCTR.ipynb [hunk 7]
```diff
@@ -140,44 +136,50 @@
...
+    "\n",
+    "Now we can save our models to be deployed at the inference stage. To do so we will use export_hugectr_ensemble method below. With this method, we can generate the config.pbtxt files automatically for each model.<br><br>\n",
```
[the remark this answers is not in the document]
Comment (bschifferer) [reply examples/scaling-criteo/04-Triton-Inference-with-HugeCTR.ipynb:152]: updated it, thanks

File: examples/scaling-criteo/04-Triton-Inference-with-HugeCTR.ipynb [hunk 8]
```diff
@@ -269,482 +272,342 @@
...
    "cell_type": "markdown",
    "metadata": {},
    "source": [
-    "We check the available models in the repositories:\n",
-    "- criteo_ens: Ensemble \n",
-    "- criteo_nvt: NVTabular \n",
-    "- criteo: HugeCTR model"
+    "## Get prediction from Triton Inference Server\n",
+    "\n",
+    "We have only saved the models for Triton Inference Server. We started Triton Inference Server and the models are loaded.  we will send raw data as a request and receive the predictions.\n",
```
[the remark this answers is not in the document]
Comment (bschifferer) [reply examples/scaling-criteo/04-Triton-Inference-with-HugeCTR.ipynb:356]: updated

File: examples/scaling-criteo/04-Triton-Inference-with-HugeCTR.ipynb [hunk 9]
```diff
@@ -269,482 +272,342 @@
...
+    "\n",
+    "We read 3 example rows from the last parquet file from the raw data. We drop the target column (`label`) from the dataframe, as the information is not available at inference time."
```
[the remark this answers is not in the document]
Comment (bschifferer) [reply examples/scaling-criteo/04-Triton-Inference-with-HugeCTR.ipynb:358]: done

File: examples/scaling-criteo/04-Triton-Inference-with-HugeCTR.ipynb [hunk 10]
```diff
@@ -269,482 +272,342 @@
...
   },
   {
    "cell_type": "markdown",
    "metadata": {},
    "source": [
-    "### Example Request to Triton Inference Server\n",
+    "We generate a Triton Inference Server request object. \n",
     "\n",
-    "Now, the models are loaded and we can create a sample request. We read an example **raw batch** for inference."
+    "Currently, NA/None values are not supported for `int32` columns. As a workaround, we will NA values with 0. This will be updated in the future. The output of the HugeCTR model is called `OUTPUT0`. For the same reason of dropping the target column, we need to remove it from the input schema, as well."
```
[the remark this answers is not in the document]
Comment (bschifferer) [reply examples/scaling-criteo/04-Triton-Inference-with-HugeCTR.ipynb:531]: done

File: examples/scaling-criteo/04-Triton-Inference-with-HugeCTR.ipynb [hunk 11]
```diff
@@ -269,482 +272,342 @@
...
-      "\u001b[0;32m/usr/local/lib/python3.8/dist-packages/tritonclient/http/__init__.py\u001b[0m in \u001b[0;36m_raise_if_error\u001b[0;34m(response)\u001b[0m\n\u001b[1;32m     62\u001b[0m     \u001b[0merror\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0m_get_error\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mresponse\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     63\u001b[0m     \u001b[0;32mif\u001b[0m \u001b[0merror\u001b[0m \u001b[0;32mis\u001b[0m \u001b[0;32mnot\u001b[0m \u001b[0;32mNone\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 64\u001b[0;31m         \u001b[0;32mraise\u001b[0m \u001b[0merror\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     65\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     66\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
-      "\u001b[0;31mInferenceServerException\u001b[0m: in ensemble 'criteo_ens', Failed to process the request(s), message: The stub process has exited unexpectedly."
+      "OUTPUT0 [0.52164096 0.50390565 0.4957397 ] (3,)\n"
```
[the remark this answers is not in the document]
Comment (bschifferer) [reply examples/scaling-criteo/04-Triton-Inference-with-HugeCTR.ipynb:587]: I added a sentence

File: examples/scaling-criteo/README.md (hunk 1 above)
Comment (bschifferer) [reply examples/scaling-criteo/README.md:20]: good catch :)

**Comment by bschifferer** [comment]
rerun tests

**Pull request merged**